### PR TITLE
feat: change release tags to vX.Y.Z. Fixes #5954

### LINKED ISSUE #5954: [backend] missing 'v' prefix in repo tags
When trying to use the go client in github.com/kubeflow/pipelines/backend/api/go_client, I find that `go mod` can't fetch the corresponding tags in this github repo, because all tags defined are missing the 'v' prefix and are considered non-canonical tag formats.

E.g., tag '1.5.1' should be 'v1.5.1'.

### PR DESCRIPTION
**Pull request by NikeNano**
**Description of your changes:**

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by NikeNano** [comment]
@Bobgy  you also mentioned that we might have to update some builds? As I understand the builds will be triggerade baed upon the version in [VERSION](https://github.com/kubeflow/pipelines/blob/master/VERSION)

File: RELEASE.md [hunk 1]
```diff
@@ -23,15 +23,15 @@ additionally.
 
 ## Release Tags and Branches
 
-Releases are tagged with tags like `X.Y.Z`, e.g. `1.0.2`. A special format like
-`1.0.0-rc.2` is a pre-release. It is the second release candidate before
-releasing the final `1.0.0`.
+Releases are tagged with tags like `v.X.Y.Z`, e.g. `v.1.0.2`. A special format like
```
Comment (Bobgy) [inline RELEASE.md:26]: I think it's more common to tag like vX.Y.Z, can you confirm?

**Comment by Bobgy** [comment]
Please search for mm_ver or mm.ver
There are some parsing logic

File: RELEASE.md (hunk 1 above)
Comment (NikeNano) [reply RELEASE.md:26]: Yes, will updated this.

**Comment by NikeNano** [comment]
> Please search for mm_ver or mm.ver
> There are some parsing logic

Found it: 
https://github.com/kubeflow/pipelines/blob/df1ab4db5e72e2ddb6f098343a3faf51599087d1/.cloudbuild.yaml#L211
https://github.com/kubeflow/pipelines/blob/df1ab4db5e72e2ddb6f098343a3faf51599087d1/.cloudbuild.yaml#L225

However I dont think we need to update it:

```
touch VERSION
echo "v1.7.0-alpha.1" > VERSION
cat VERSION | sed -e "s#\([0-9]\+[.][0-9]\+\)[.].*#\1#"
```
gives : 

```
v1.7
```

Where the expected out put should follow: 

```
 # Parse major minor version and save to a file for reusing in other steps.
 # e.g. 1.0.0-rc.1 and 1.0.1 are parsed as 1.0
```
 and thus I think it should be fine.

**Comment by Bobgy** [comment]
The mm_ver is used in GCP marketplace.
I am not sure whether it supports adding the v prefix.

We can try

File: test/release/README.md [hunk 2]
```diff
@@ -4,18 +4,18 @@ For full release documentation, please read [RELEASE.md](../../RELEASE.md).
...
 To do a release:
 
-TAG=`<TAG>` BRANCH=`<BRANCH>` make release
+TAG=`<TAG>` BRANCH=`<BRANCH>` make release, the release needs to start with `v` to support [golang version handling](https://golang.org/ref/mod#versions)
 
 Example, release in release branch:
 
 ```bash
-    TAG=1.4.0 BRANCH=release-1.4 make release
+    TAG=v.1.4.0 BRANCH=release-1.4 make release
```
Comment (Bobgy) [inline test/release/README.md:12]: Nit update

File: test/release/README.md [hunk 3]
```diff
@@ -4,18 +4,18 @@ For full release documentation, please read [RELEASE.md](../../RELEASE.md).
...
 ```
 
 Example, release an RC (release candidate) in master branch:
 
 ```bash
-    TAG=2.0.0-rc.1 BRANCH=master make release
+    TAG=v.2.0.0-rc.1 BRANCH=master make release
```
Comment (Bobgy) [inline test/release/README.md:18]: Nit update

File: test/release/release.sh [hunk 4]
```diff
@@ -20,6 +20,12 @@ TAG=$1
 BRANCH=$2
 REPO=kubeflow/pipelines
 
+if  [[ $TAG != v* ]] ;
+then
+    echo "release tag needs to start with 'v' "
```
Comment (Bobgy) [inline test/release/release.sh:25]: Thanks for checking for validity!

File: test/release/release.sh [hunk 5]
```diff
@@ -37,7 +43,7 @@ cd "$clone_dir"
 git checkout "$BRANCH"
 
 echo "Preparing local git tags used by changelog generation."
-# tags with "-" are pre-releases, e.g. 1.0.0-rc.1
+# tags with "-" are pre-releases, e.g. v.1.0.0-rc.1
```
Comment (Bobgy) [inline test/release/release.sh:46]: Nit update

**Review by Bobgy — COMMENTED** [review]
Thank you! Some minor clean up left.
Otherwise
/lgtm

**Comment by Bobgy** [comment]
/lgtm
/approve

Thank you for fixing this for go modules!

**Comment by capri-xiyue** [comment]
I encountered  error`Step #49 - "tagCacheServerForMarketplace": Error parsing reference: "gcr.io/ml-pipeline/google/pipelines/cacheserver:" is not a valid repository/tag: invalid reference format` when I tried to make a release.

**Pull request merged**